packages/@sanity/form-builder/src/inputs/ObjectInput/ObjectInput.tsx

```diff
@@ -74,8 +79,24 @@ export const ObjectInput = memo(
     } = props
 
     const handleFieldChange = React.useCallback(
-      (fieldEvent: PatchEvent, field) => {
+      (fieldEvent: PatchEvent, field: ObjectField) => {
         let event = fieldEvent.prefixAll(field.name)
+
+        // check if the result of the incoming patches will result in an empty object.
+        // first apply all the patches to the current value to get the result.
+        const result = applyAll(
```

Comment:
I'm a bit concerned about potential performance penalties with doing this, especially for patches that targets deeply nested objects. That said, I did some quick testing without noticing any significant difference, so I guess it's fine. Something to keep an eye out for nevertheless.

Reply:
i think that's a good point. i could probably check if there is an `unset` in the chain and bail out if that's not found. i think that works out for the most part